cli/bpmetadata/cmd.go

```diff
@@ -233,23 +231,23 @@ func createInfo(bpPath string, readmeContent []byte) (*BlueprintInfo, error) {
 		i.Icon = iconFilePath
 	}
 
-	return i, nil
+	return nil
 }
 
-func createInterfaces(bpPath string, interfaces *BlueprintInterface) (*BlueprintInterface, error) {
+func (i *BlueprintInterface) create(bpPath string) error {
 	i, err := getBlueprintInterfaces(bpPath)
```

Comment:
Since the pointer receiver is also `i`, wouldn't this shadow it rather than assign it?
I am thinking something like https://go.dev/play/p/iZEdjIAx3Dd may happen.

Reply:
Yeah just realized that. We'll have to do something like this: https://go.dev/play/p/N5KOguNzOHz. I've changed it.

Reply:
@g-awmalik In a follow up, could we also add some e2e tests which can catch these?